src/lavinmq/deduplication.cr

```diff
@@ -0,0 +1,66 @@
+module LavinMQ
+  module Deduplication
+    abstract class Cache(T)
+      abstract def contains?(key : T) : Bool
+      abstract def insert(key : T, ttl : UInt32?)
+    end
+
+    class MemoryCache(T) < Cache(T)
+      def initialize(@size : UInt32)
+        @store = Hash(T, Int64?).new
+      end
```

Comment:
isn't the purpose of the TTL is to keep the size of the cache reasonable small? So I think you should have a fiber that checks the TTL of each key and evict the ones that has expired. Not check the TTL in the `contains?` method. 

monotonic is a better alternative when you only are interested in the timespan, not the exact point of time. 

```suggestion
    class MemoryCache(T) < Cache(T)
      def initialize(@size : UInt32)
        @store = Hash(T, Int64?).new
        spawn(name: "memorycache expiry") do
          loop do
            sleep 60.seconds
            @store.reject! { |_, ttd| tdd.nil? || RoughTime.monotonic > ttd }
          end
        end
```

Reply:
thinking more about it, maybe i'm wrong about the purpose of the TTL here.